sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,443 @@
+      sessionId: getStringOrUndefined(value["SessionId"]),
+      messageId: getStringOrUndefined(value["MessageId"]),
+      contentType: getStringOrUndefined(value["ContentType"]),
+      userProperties: value["UserProperties"]
+    };
+  }
+  return result;
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve rule `action` value from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+function getRuleActionOrUndefined(value: any): SqlAction | undefined {
+  if (value == undefined) {
+    return undefined;
+  } else {
+    return {
+      sqlExpression: value["SqlExpression"],
+      sqlParameters: getSqlParametersOrUndefined(value["Parameters"]),
+      compatibilityLevel: getIntegerOrUndefined(value["CompatibilityLevel"]),
+      requiresPreprocessing: getBooleanOrUndefined(value["RequiresPreprocessing"])
+    };
+  }
+}
+
+/**
+ * Represents settable options on a rule
+ */
+export interface RuleOptions {
+  /**
+   * Defines the expression that the rule evaluates. The expression string is interpreted as a SQL92 expression which must evaluate to True or False. Only one between a correlation and a sql expression can be defined.
+   */
+  filter?: SqlFilter | CorrelationFilter;
+
+  /**
+   * The SQL like expression that can be executed on the message should the associated filter apply.
+   */
+  action?: SqlAction;
+}
+
+/**
+ * @ignore
+ * Internal representation of settable options on a rule
+ */
+export interface InternalRuleOptions extends RuleOptions {
+  /**
+   * Name of the rule.
+   */
+  name?: string;
+}
+
+/**
+ * Represents all attributes of a rule entity
+ */
+export interface Rule extends RuleOptions {
+  /**
+   * Name of the rule
+   */
+  ruleName?: string;
+
+  /**
+   * Name of topic
+   */
+  topicName?: string;
+
+  /**
+   * Name of subscription
+   */
+  subscriptionName?: string;
+
+  /**
+   * Created at timestamp
+   */
+  createdAt?: string;
+}
+
+/**
+ * Represents all possible fields on SqlAction
+ */
+export type SqlAction = SqlFilter;
+
+/**
+ * Represents all possible fields on SqlFilter
+ */
+export interface SqlFilter {
+  /**
+   * SQL expression to use.
+   */
+  sqlExpression?: string;
+
+  /**
+   * SQL parameters to the expression
+   */
+  sqlParameters?: SqlParameter[];
+
+  /**
+   * This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20.
+   */
+  compatibilityLevel?: number;
+
+  /**
+   * Boolean value indicating whether the SQL filter expression requires preprocessing
+   */
+  requiresPreprocessing?: boolean;
+}
+
+/**
+ * @ignore RuleResourceSerializer for serializing / deserializing Rule entities
+ */
+export class RuleResourceSerializer implements AtomXmlSerializer {
+  serialize(rule: InternalRuleOptions): object {
+    const resource: { Name: any; Filter: any; Action: any } = {
+      Filter: {},
+      Action: {},
+      Name: rule.name
+    };
+
+    if (rule == undefined || rule.filter == undefined) {
```

Comment:
I believe the `rule == undefined` check is not required here.
serialize() is called on an object that we create which in the worse case scenario would be an empty object.

Also, a few lines above we are accessing `rule.name`. So doing a `rule == undefined` check after that would not yield anything

Reply:
Updated